hpe3parclient/client.py

```diff
@@ -4816,3 +4731,159 @@ def getScheduleStatus(self, schedule_name):
+                    {
+                        'Package': 'Complete',
+                        'Version': '3.1.3.230',
+                        'InstallTime': '2014-06-06 14:46:56 PDT',
+                        'Id': 'MU1'
+                    }
+                ]
+            }
+
+            # Example 2: No data rows, so zero members.
+            ret = {'total': 0, 'members': []}
+
+        """
+
+        members = []
+        if cli_output and len(cli_output) >= 2:
+            for index, line in enumerate(cli_output):
+                if index == 0:
+                    headers = line.split(',')
+                else:
+                    split = line.split(',')
+                    member = {}
+                    for i, header in enumerate(headers):
+                        try:
+                            member[header] = split[i]
+                        except IndexError:
+                            member[header] = None
+                    members.append(member)
+
+        return {'total': len(members), 'members': members}
+
+    @staticmethod
+    def getPortNumber(ip, login, password, port=22,
+                      conn_timeout=None, privatekey=None,
+                      **kwargs):
+        """Get port number from showwsapi output
+
+        :param 3PAR credentials
+        :return: HTTPS_Port column value
+        """
+        ssh_client = ssh.HPE3PARSSHClient(ip, login, password, port,
+                                          conn_timeout, privatekey,
+                                          **kwargs)
+        if ssh_client is None:
+            raise exceptions.SSHException('SSH is not initialized. Initialize'
+                                          ' it by calling "setSSHOptions".')
+        ssh_client.open()
+        cli_output = ssh_client.run(['showwsapi'])
+        if ssh_client:
+            ssh_client.close()
+        wsapi_collection = HPE3ParClient.convert_cli_output_to_wsapi_format(
+            cli_output)
+        return wsapi_collection['members'][0]['HTTPS_Port']
+
+    def tuneVolume(self, volName, tune_operation, optional=None):
+        info = {'action': self.TUNE_VOLUME, 'tuneOperation': tune_operation}
+        if optional is not None and not self.compression_supported:
+            if 'compression' in optional.keys():
+                del optional['compression']
+        if optional:
+            if self.primera_supported:
+                msg = "invalid input: For Deco volumes 'conversionOperation' "\
+                      "should be 3(TDVV) and 'compression' must be specified"\
+                      " as true"
+                msg1 = "invalid input:'conversionOperation' value 2(FPVV) is "\
+                       "not supported"
+                if optional.get('conversionOperation') == 3 \
+                        and optional.get('compression') is True:
+                    optional['conversionOperation'] = 4
+
+                if optional.get('conversionOperation') == 3 \
```

Comment:
@ansarars Please do
if (optional.get('conversionOperation') == 1 and optional.get('compression') is True) or (optional.get('conversionOperation') == 3 and optional.get('compression') is False):
raise exceptions.HTTPBadRequest("invalid input: For Deco volumes 'conversionOperation' "\
                      "should be 3(TDVV) and 'compression' must be specified"\
                      " as true")

You can then remove lines 4880-4882 and 4864-4866.